balancer/base/balancer.go

```diff
@@ -129,22 +149,26 @@ func (b *baseBalancer) UpdateClientConnState(s balancer.ClientConnState) error {
 
 // regeneratePicker takes a snapshot of the balancer, and generates a picker
 // from it. The picker is
-//  - errPicker with ErrTransientFailure if the balancer is in TransientFailure,
+//  - errPicker if the balancer is in TransientFailure,
 //  - built by the pickerBuilder with all READY SubConns otherwise.
-func (b *baseBalancer) regeneratePicker(err error) {
+func (b *baseBalancer) regeneratePicker() {
 	if b.state == connectivity.TransientFailure {
 		if b.pickerBuilder != nil {
 			b.picker = NewErrPicker(balancer.ErrTransientFailure)
 		} else {
-			if err != nil {
-				b.v2Picker = NewErrPickerV2(balancer.TransientFailureError(err))
-			} else {
-				// This means the last subchannel transition was not to
-				// TransientFailure (otherwise err must be set), but the
-				// aggregate state of the balancer is TransientFailure, meaning
-				// there are no other addresses.
-				b.v2Picker = NewErrPickerV2(balancer.TransientFailureError(errors.New("resolver returned no addresses")))
+			// Build the error from the last connection error and the last
+			// resolver error.  connErr must always be non-nil unless there are
+			// no SubConns, in which case resolverErr must be non-nil.
+			err := b.connErr
```

Comment:
Not sure if this is any better

```go
var desc string
if err := b.connErr; err != nil {
  desc += err.Error()
}
if err := b.resolverErr; err != nil {
  desc += err.Error()
}
b.v2Picker = NewErrPickerV2(balancer.TransientFailureError(fmt.Errorf(desc)))
```

Reply:
I'm not sure I like the string manipulation when there is only one non-nil error (the norm).  How about this?